samza-test/src/test/java/org/apache/samza/test/processor/TestZkStreamProcessor.java

```diff
@@ -0,0 +1,558 @@
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.samza.test.processor;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.Map;
+import java.util.Properties;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import kafka.utils.TestUtils;
+import org.apache.kafka.clients.consumer.ConsumerRecord;
+import org.apache.kafka.clients.consumer.ConsumerRecords;
+import org.apache.kafka.clients.consumer.KafkaConsumer;
+import org.apache.kafka.clients.producer.KafkaProducer;
+import org.apache.kafka.clients.producer.ProducerRecord;
+import org.apache.samza.config.ApplicationConfig;
+import org.apache.samza.config.Config;
+import org.apache.samza.config.JobCoordinatorConfig;
+import org.apache.samza.config.MapConfig;
+import org.apache.samza.config.ZkConfig;
+import org.apache.samza.processor.StreamProcessor;
+import org.apache.samza.processor.StreamProcessorLifecycleListener;
+import org.apache.samza.system.IncomingMessageEnvelope;
+import org.apache.samza.system.OutgoingMessageEnvelope;
+import org.apache.samza.system.SystemStream;
+import org.apache.samza.task.InitableTask;
+import org.apache.samza.task.MessageCollector;
+import org.apache.samza.task.StreamTask;
+import org.apache.samza.task.TaskContext;
+import org.apache.samza.task.TaskCoordinator;
+import org.apache.samza.test.StandaloneIntegrationTestHarness;
+import org.apache.samza.test.StandaloneTestUtils;
+import org.apache.samza.zk.TestZkUtils;
+import org.junit.Assert;
+import org.junit.Test;
+
+public class TestZkStreamProcessor extends StandaloneIntegrationTestHarness {
+  /**
+   * Testing a basic identity stream task - reads data from a topic and writes it to another topic
+   * (without any modifications)
+   *
+   * <p>
+   * The standalone version in this test uses KafkaSystemFactory and it uses a SingleContainerGrouperFactory. Hence,
+   * no matter how many tasks are present, it will always be run in a single processor instance. This simplifies testing
+   */
+
+  public static StreamProcessorLifecycleListener listener = new StreamProcessorLifecycleListener() {
+    @Override
+    public void onStart() {
+    }
+
+    @Override
+    public void onShutdown() {
+    }
+
+    @Override
+    public void onFailure(Throwable t) {
+    }
+  };
+
+  @Test
+  public void testSingleStreamProcessor() {
+    testStreamProcessor(new String[]{"1"});
+  }
+
+  @Test
+  public void testTwoStreamProcessors() {
+    testStreamProcessor(new String[]{"1", "2"});
+  }
+
+  @Test
+  public void testFiveStreamProcessors() {
```

Comment:
Can you split the tests into "setup" , "test" and "teardown" sections using @Before and @After annotations? It will be much cleaner and safer to run. 

Looking at this testStreamProcessor line here, I can't tell what your test is and what your assertions are.